### PR TITLE
1.3/develop

### PR DESCRIPTION
**Pull request by pbarclay**
While dealing with collisions on migration script numbers across branches it seemed to be a good idea to change the migration number to use a time stamp which should ensure the number is consecutive, current and unique to another developer creating migrations when on another branch.

### CODE REVIEW

**Comment by WanWizard** [comment]
@philsturgeon can you live with this?

**Comment by philsturgeon** [comment]
The trouble here is that its not a full solution, you still have to make sure that each feature branch is merged and run in the exact right order or you'll end up having migrations missed!

What you need to do is track ALL numbers in the database, then the timestamp works. But that of course can lead to issues where earlier folks have done something drastically different to what a later migration expects. That sadly is always going to happen and cannot be avoided.

To summarise my rambling, this is a good change BUT only if the migrations are all tracked. Just storing the latest number when that number is a timestamp, as people get lazy - forget they need to worry about migrations and generally f*_k sh_t up.

So as it is, no. With some extra work, yes.

**Comment by pbarclay** [comment]
I see your point - I will look at extending the migration table and decoupling the file name from the order of execution and see where that leads - the time stamp can still be useful if 2 branches each create an "alter_user_table" migration. By looking at the migrations and comparing that with the migrations that have been run in the database migration table then it should be possible to determine what migrations have not been run if not the order they should be run so that and allow an execution order to be specified once a merge of feature branches has been done.

Anyway thanks for the feedback.
